DoggyFriction/Services/DebtAggregator.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Ajax.Utilities;
+
+namespace DoggyFriction.Services
+{
+    public class DebtAggregator
+    {
+        private class DebtAggregatorKey : IEquatable<DebtAggregatorKey>
+        {
+            public string Creditor { get; }
+            public string Debtor { get; }
+
+            public override int GetHashCode()
```

Comment:
Maybe it's better to use Resharper's implementation of `GetHashCode` and `Equals`. Try to hit ALT+INS and generate equality members